vulture_os/toolkit/api_parser/cybereason/cybereason.py

```diff
@@ -321,49 +371,51 @@ def suspicionsNetwork(self, alertId):
-        if (data['status'] != 'SUCCESS'):
+        if data['status'] != 'SUCCESS':
             return False, data['message']
         return True, data
 
-    def getDevicesDetails(self, devicesIds):
+    def get_devices_details(self, devices_names):
         sensors_url = f"{self.host}/{self.SENSOR_URI}"
         query = {
             'sortDirection': 'ASC',
             'sortingFieldName': 'machineName',
             'filters': [
-                {'fieldName': "machineName", 'operator': "ContainsIgnoreCase", 'values': devicesIds},
+                {'fieldName': "machineName", 'operator': "ContainsIgnoreCase", 'values': devices_names},
                 {"fieldName": "status", "operator":"NotEquals", "values":["Archived"]}
             ]
         }
         data = self.execute_query("POST", sensors_url, query)
         return data.get('sensors', [])
 
-    def format_log(self, log):
+    @staticmethod
+    def format_log(kind, log):
 
-        log['iconBase64'] = ""
+        if kind == "malops":
+            log['iconBase64'] = ""
 
-        flattened_ip = []
-        flattened_hostname = []
-        flattened_domain = []
+            flattened_ip = []
+            flattened_hostname = []
+            flattened_domain = []
 
-        for device in log['devices']:
-            if device['ip'] not in flattened_ip:
-                flattened_ip.append(device['ip'])
-            if device['nat']['ip'] not in flattened_ip:
-                flattened_ip.append(device['nat']['ip'])
-            if device['hostname'] not in flattened_hostname:
-                flattened_hostname.append(device['hostname'])
-            if device['domain'] not in flattened_domain:
-               flattened_domain.append(device['domain'])
+            for device in log['devices']:
+                if device['ip'] not in flattened_ip:
+                    flattened_ip.append(device['ip'])
+                if device['nat']['ip'] not in flattened_ip:
+                    flattened_ip.append(device['nat']['ip'])
+                if device['hostname'] not in flattened_hostname:
+                    flattened_hostname.append(device['hostname'])
+                if device['domain'] not in flattened_domain:
+                   flattened_domain.append(device['domain'])
 
-        for user in log['users']:
-            domain = (user['displayName']).split('\\')[0]
-            if domain not in flattened_domain:
-                flattened_domain.append(domain)
+            for user in log['users']:
+                domain = (user['displayName']).split('\\')[0]
+                if domain not in flattened_domain:
+                    flattened_domain.append(domain)
 
-        log['flattened_ip'] = flattened_ip
-        log['flattened_hostname'] = flattened_hostname
-        log['flattened_domain'] = flattened_domain
+            log['flattened_ip'] = flattened_ip
+            log['flattened_hostname'] = flattened_hostname
+            log['flattened_domain'] = flattened_domain
```

Comment:
`related` root object exists in ECS, is that taken into account?

Reply:
Yes, flattened fields is mapped to `related` ECS object

Reply:
Yes these fields are mapped to `related` ECS object